Dev login

christianadlaon opened the pull request.

christianadlaon committed:
Initial commit

Set up project structure and add initial files.

christianadlaon committed:
Fix: Remove generated files and configure line endings

christianadlaon committed:
Add new feature to process user input

Implemented a function to handle and validate user input in the main application module. This change improves input reliability and prepares the codebase for future enhancements.

christianadlaon committed:
Implement Bloc-based Firebase authentication

Introduces Bloc and Cubit architecture for Firebase authentication, including AuthBloc, LoginFormCubit, SignupFormCubit, and repository pattern. Updates main.dart to provide Bloc instances globally and refactors auth_wrapper.dart to use Bloc state. Adds comprehensive documentation and an example Bloc-based login screen. Removes the old mock AuthService.

melchorlapinigjr commented on lib/screens/signup_screen.dart:
```diff
@@ -0,0 +1,575 @@
+          backgroundColor: Colors.red,
+          duration: Duration(seconds: 2),
+        ),
+      );
```
Create a single method that accepts context, message, and color (nullable) to be used in all occurences.
Something like:
`void toastMessage({ required String message,required BuildContext context, Color? color})(){
ScaffoldMessenger.of(context).showSnackBar(.....);
}`

and call toastMessage()... anywhere in this class.

melchorlapinigjr commented on lib/screens/bloc_login_screen.dart:
Rename the name of this file to login_screen.dart only since this only holds widgets, not logics.

melchorlapinigjr commented on lib/screens/bloc_login_screen.dart:
```diff
@@ -0,0 +1,430 @@
+// import 'signup_screen.dart';
+
+// class BlocLoginScreen extends StatefulWidget {
+//   const BlocLoginScreen({super.key});
```
Remove Bloc in the class name.

melchorlapinigjr commented on lib/controllers/login_controller.dart:
```diff
@@ -0,0 +1,146 @@
+import '../services/firebase_auth_service.dart';
+import '../utils/auth_debug.dart';
+
+class LoginController {
```
Where is this **LoginController**  eing used? The methods inside here can be moved to login_bloc or auth_bloc.

christianadlaon committed:
Remove Bloc authentication guide and refactor auth code

Deleted documentation files related to Bloc authentication and hot reload logout features. Removed the Bloc-based login screen and refactored authentication, login, and signup logic for improved readability and maintainability. Introduced a reusable toast message method in login and signup screens for consistent user feedback.
